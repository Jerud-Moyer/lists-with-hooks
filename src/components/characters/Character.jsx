import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image }) => (
  <>
    <img src={image} alt={name} />
    <h1>{name}</h1>
  </>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Character;
