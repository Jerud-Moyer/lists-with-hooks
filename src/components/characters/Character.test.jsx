import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  afterEach(() => cleanup());

  it('renders Character', () => {
    const { asFragment } = render(<Character
      image="Test Image"
      name="test Nameu"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
