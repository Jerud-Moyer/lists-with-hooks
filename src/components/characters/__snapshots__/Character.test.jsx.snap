// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <img
    alt="test Nameu"
    src="Test Image"
  />
  <h1>
    test Nameu
  </h1>
</DocumentFragment>
`;
